### PR TITLE
Some C++ fixes

### PR DESCRIPTION
**Pull request by jacquesg**
This adds ```-fno-exceptions``` for C++ files and makes ```-fpack-struct``` only applicable to C files.

### CODE REVIEW

**Comment by NicoHood** [comment]
Why dont you want to use `-fpack-struct` in c++ files?

**Comment by jacquesg** [comment]
You very quickly end up with:

```warning: ignoring packed attribute because of unpacked non-POD field```

In general, IMHO packing a struct should be done on a per struct basis. I understand that this is typically what you want for a small MCU's, but it does make it hard to mix static libraries built with another build system due to the (potential) ABI difference.

**Comment by NicoHood** [comment]
@jacquesg that makes sense to me. However who uses unpacked structs? Does it even make sense to not pack structs in c++? In general I'd agree to your argument, but on avr I dont see why you not want to pack structs, especially because we are on an 8 bit system.

**Comment by NicoHood** [comment]
How about also adding `-fno-threadsafe-statics` to CPP_FLAGS? From what I saw this really makes sense. The option also safes space for my code.

File: DMBS/gcc.mk [hunk 1]
```diff
@@ -26,7 +26,7 @@ F_CPU              ?=
 C_STANDARD         ?= gnu99
 CPP_STANDARD       ?= gnu++98
 C_FLAGS            ?=
-CPP_FLAGS          ?=
+CPP_FLAGS          ?= -fno-exceptions
```
Comment (NicoHood) [inline DMBS/gcc.mk]: I noticed that this location is actually wrong. It needs to move to BASE_CPP_FLAGS instead.

**Comment by jacquesg** [comment]
Sounds fair, updated.

**Comment by NicoHood** [comment]
nice :)

**Pull request merged**